src/app/transactional-search/transactional-search.component.spec.ts

```diff
@@ -0,0 +1,167 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { TransactionalSearchComponent } from './transactional-search.component';
+import { TransactionSearchService } from '../transaction-search.service';
+import { ITransactionDetail } from './transaction.model';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+import { HttpClientModule } from '@angular/common/http';
+
+const mockTransactionList: ITransactionDetail[] = [
+ {
+  Description: 'Installment',
+  Date: '20Nov2017',
+  Amount: -2000.00,
+  Balance: 13951.84
+ },
+ {
+  Date: '15Oct2017',
+  Description: 'Purchase ',
+  Amount: 4800.00,
+  Balance: 13951.84}
+];
+
+
+const mockTransactionListSecond: ITransactionDetail[] = [
+  {
+    Date: '15Oct2017',
+    Description: 'Purchase ',
+    Amount: 4800.00,
+    Balance: 13951.84},
+  {
+    Description: 'Installment',
+   Date: '20Nov2017',
+   Amount: -2000.00,
+   Balance: 13951.84
+  }];
+
+
+const mockTransactionServiceStub = {
+  getData: jasmine.createSpy('getData').and.returnValue(Observable.of(mockTransactionList))
+};
+
+const mockSortType = 'Amount';
+
+// const error = Observable.create(observer => {
+//   observer.error(new Error('error'));
+//   observer.complete();
+// });
+
+describe('TransactionalSearchComponent', () => {
+  let component: TransactionalSearchComponent;
```

Comment:
use const

Reply:
In this file component can be use many times it can not be const. Hence I keep it as it is.